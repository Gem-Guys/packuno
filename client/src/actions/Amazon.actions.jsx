const rp = require('request-promise');

const setAmazonData = result => ({
  type: 'SET_AMAZON_DATA',
  data: result,
});

const retrieveAmazonAsync = query => (dispatch) => {
  console.log('-------------------->', query);
  const options = {
    type: 'GET',
    uri: 'http://localhost:3000/amazon',
    qs: { query },
  };
  const AmazonItems = rp(options);
  return AmazonItems.then((result) => {
    console.log('api results ------>', result);
    dispatch(setAmazonData(result));
  });
};

module.exports.retrieveAmazonAsync = retrieveAmazonAsync;
