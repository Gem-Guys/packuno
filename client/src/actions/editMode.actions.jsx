export const enterEditItemMode = (id) => {
  return {
    type: 'ENTER_EDIT_ITEM_MODE',
    id,
  };
};

export const exitEditItemMode = () => {
  return {
    type: 'EXIT_EDIT_ITEM_MODE',
  };
};
