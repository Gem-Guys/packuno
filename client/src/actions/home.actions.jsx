
let itemID = 0;

export const addItem = (item, category) => {
  return {
    type: 'ADD_ITEM',
    id: itemID++,
    item, 
    category,
    packed: false
  }
};


export const togglePacked = (id) => {
  return {
    type: 'TOGGLE_PACKED',
    id
  }
};


export const setCatFilter = (selectedCategory) => {
  return {
    type: 'SET_CAT_FILTER',
    selectedCategory
  }
};

