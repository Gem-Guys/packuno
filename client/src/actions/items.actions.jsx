
let itemID = 4;

export const addItem = (item, category = 'Other', quantity = 0) => {
  return {
    type: 'ADD_ITEM',
    id: itemID += 1,
    item,
    category,
    quantity,
    packed: false,
  };
};


export const togglePacked = (id) => {
  return {
    type: 'TOGGLE_PACKED',
    id,
  };
};

export const deleteItem = (id) => {
  return {
    type: 'DELETE_ITEM',
    id,
  };
};

export const editItem = (id, item, category, quantity) => {
  return {
    type: 'EDIT_ITEM',
    id,
    item,
    category,
    quantity,
  };
};


export const setVisibilityFilter = (filter) => {
  return {
    type: 'SET_VISIBILITY_FILTER',
    filter,
  };
};
