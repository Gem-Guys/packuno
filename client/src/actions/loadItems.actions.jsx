// const request = require('request');
const axios = require('axios');

export function userItemsHasErrored(bool) {
  console.log('in userItemsHasErrored');
  return {
    type: 'USERITEMS_HAS_ERRORED',
    hasErrored: bool,
  };
}

export function userItemsIsLoading(bool) {
  return {
    type: 'USERITEMS_IS_LOADING',
    isLoading: bool,
  };
}

export function userItemsFetchDataSuccess(data) {

  const userItems = {};
  data.forEach(userItem => userItems[userItem.item] = userItem.category);

  return {
    type: 'USERITEMS_FETCH_DATA_SUCCESS',
    userItems,
  };
}

export function userItemsFetchData() {
  return (dispatch) => {
    dispatch(userItemsIsLoading(true));

    axios.get('/userItems')
      .then((response) => {
        dispatch(userItemsIsLoading(false));
        dispatch(userItemsFetchDataSuccess(response.data));
      })
      .catch(err => dispatch(userItemsHasErrored(true)));
  };
}