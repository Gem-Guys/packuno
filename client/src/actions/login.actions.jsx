const request = require('request');
const rp = require('request-promise');

export const loggedIn = bool => ({
  type: 'LOG_IN',
  loggedIn: bool,
});

export const loggedInAsync = () => (dispatch, getState) => {
  const options = {
    type: 'GET',
    uri: 'http://localhost:3000/check/',
  };
  rp(options).then((result) => {
    dispatch(loggedIn(result.toString()));
  });
};

export const currentUser = userID => ({
  type: 'SET_CURRENT_USER',
  userId: userID,
});

export const currentUserAsync = userID => (dispatch, getState) => {
  const options = {
    type: 'GET',
    uri: 'http://localhost:3000/user/',
  };
  rp(options).then((result) => {
    const id = JSON.parse(result);
    return id.id;
  }).then((id) => {
    dispatch(currentUser(id));
  });
};

