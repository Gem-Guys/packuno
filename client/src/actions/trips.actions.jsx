const axios = require('axios');

// Get user's previous trips

// export const tripsHasErrored = bool => ({
//   type: 'TRIPS_HAS_ERRORED',
//   hasErrored: bool,
// });

// export const tripsIsLoading = bool => ({
//   type: 'TRIPS_IS_LOADING',
//   isLoading: bool,
// });

// export const itemsFetchDataSuccess = (data) => {
//   const items = {
//     byId: {},
//     allIds: [],
//     categories: [],
//   };

//   data.forEach((tripItem) => {
//     const { id, Item, packed, quantity, itemId } = tripItem;

//     items.byId[id] = {
//       id,
//       item: Item.item,
//       category: Item.category,
//       quantity,
//       packed,
//       itemId,
//     };

//     items.allIds.push(id);
//     if (!items.categories.includes(Item.category)) {
//       items.categories.push(Item.category);
//     }
//   });

//   return {
//     type: 'ITEMS_FETCH_DATA_SUCCESS',
//     items,
//   };
// };

// export const itemsFetchData = tripId => (dispatch) => {
//   dispatch(itemsIsLoading(true));

//   axios.get(`/trips/${tripId}/items`)
//     .then((response) => {
//       dispatch(itemsIsLoading(false));
//       dispatch(itemsFetchDataSuccess(response.data));
//     })
//     .catch(() => dispatch(itemsHasErrored(true)));
// };







export const addTrip = (destination, startDate, endDate, oldTripId, userId) => (dispatch) => {
  axios.post('/trips', {
    destination,
    start_date: startDate,
    end_date: endDate,
    userId,
  })
    .then((response) => {
      dispatch({
        type: 'ADD_TRIP_SUCCESS',
        id: response.data.id,
        destination,
        startDate,
        endDate,
        oldTripId,
        userId,
      });

      if (oldTripId) {
        axios.post('/trips/items', {
          tripId: response.data.id,
          oldTripId,
        })
          .catch(err => console.log(err));
      }
    });
};

// export const selectOldTrip = (oldTripId) => {
//   return {
//     type: 'SELECT_TRIP',
//     oldTripId: oldTripId,
//   };
// };
