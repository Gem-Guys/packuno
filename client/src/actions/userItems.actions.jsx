const axios = require('axios');


// Get User's past items so they are available in autocomplete with the most recently used category

export function userItemsHasErrored(bool) {
  return {
    type: 'USERITEMS_HAS_ERRORED',
    hasErrored: bool,
  };
}

export function userItemsIsLoading(bool) {
  return {
    type: 'USERITEMS_IS_LOADING',
    isLoading: bool,
  };
}

export function userItemsFetchDataSuccess(data) {
  const userItems = {};
  data.forEach(userItem => userItems[userItem.item] = userItem.category);

  return {
    type: 'USERITEMS_FETCH_DATA_SUCCESS',
    userItems,
  };
}

export function userItemsFetchData(userId) {
  return (dispatch) => {
    dispatch(userItemsIsLoading(true));

    axios.get(`/users/${userId}/items`)
      .then((response) => {
        dispatch(userItemsIsLoading(false));
        dispatch(userItemsFetchDataSuccess(response.data));
      })
      .catch(() => dispatch(userItemsHasErrored(true)));
  };
}
