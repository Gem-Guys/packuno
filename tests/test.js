import 'babel-polyfill';
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import WeatherCont from '../client/src/containers/Weather.container';
import { Provider } from 'react-redux';
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { render } from 'react-dom';
import thunk from 'redux-thunk';
import reducer from '../client/src/reducers/index.reducers';
import Trip from '../client/src/components/Trip.component';
import Dashboard from '../client/src/components/Dashboard.component';
import LoginCont from '../client/src/containers/Login.container';
import App from '../client/src/components/App.components';
import { loggedIn, currentUser } from '../client/src/actions/login.actions';
import { setHistoricalAsync as Historical, setForecastAsync as Forecast } from '../client/src/actions/weather.actions';
import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f,
  ),
);
describe('should work as a testing suite', () => {

  test('should be able to run tests', () => {
    expect(1 + 2).toEqual(3);
  });
});

describe('should render components as pure functions', () => {
  test('renders login correctly', () => {
    const tree = renderer.create(<Provider store={store}><LoginCont /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('should create an action', () => {
  const loggedInTest = {
    type: 'LOG_IN',
    loggedIn: 'true',
  };
  expect(loggedIn('true')).toEqual(loggedInTest);
});

describe('should create actions when passed correct data', () => {
  test('should create an action when passed a login', () => {
    const loggedInTest = {
      type: 'LOG_IN',
      loggedIn: 'true',
    };
    expect(loggedIn('true')).toEqual(loggedInTest);
  });

  test('should create an action when passed historical data', () => {
    const expected = {
      type: 'SET_CURRENT_USER',
      userId: 11,
    };
    expect(currentUser(11)).toEqual(expected);
  });
});

describe('should dispatch reducers', () => {
  test('should return the initial state', () => {
    expect(reducer(undefined, {}).currentTripId).toEqual(null);
  });
  test('should return the initial state', () => {
    expect(reducer(undefined, {}).currentTripId).toEqual(null);
  });
  test('reducer should have combined actions', () => {
    expect(Object.keys(reducer(undefined, {})).length).toEqual(17);
});
});

describe('async actions', () => {
  afterEach(() => {
    fetchMock.reset()
    fetchMock.restore()
  })
  test('returns array with length greater than 0 when fetching information has been done', () => {
    const expected = [ { type: 'SET_HISTORICAL', historical: '[["August",65],["September",60]]' } ]
    const store = mockStore({ setHistorical: [] })
    return store.dispatch(Historical()).then(() => {
      expect(store.getActions().length).toBeGreaterThan(0);
    });
  });
  test('returns array with length greater than 0 when fetching information has been done', () => {
    const expected = [ { type: 'SET_FORECAST', historical: [] } ]
    const store = mockStore({ Forecast: [] })
    return store.dispatch(Forecast()).then(() => {
      expect(store.getActions().length).toBeGreaterThan(0);
    });
  });
});
