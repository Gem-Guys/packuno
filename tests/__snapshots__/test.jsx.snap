// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="weather-widget"
  />
</div>
`;

exports[`renders login correctly 1`] = `
<div>
  <a
    href="/auth/google"
  >
    Log In With Google
  </a>
</div>
`;

exports[`renders weather correctly 1`] = `
<div>
  <div
    className="weather-widget"
  />
</div>
`;
